prow/github/client.go

```diff
@@ -89,19 +90,36 @@ func (c *Client) log(methodName string, args ...interface{}) {
 	c.Logger.Printf("%s(%s)", methodName, strings.Join(as, ", "))
 }
 
-// Retry on transport failures. Does not retry on 500s.
+var timeSleep = time.Sleep
+
+// Retry on transport failures. Retries on 500s and retries after sleep on
+// ratelimit exceeded.
 func (c *Client) request(method, path string, body interface{}) (*http.Response, error) {
 	var resp *http.Response
 	var err error
 	backoff := retryDelay
 	for retries := 0; retries < maxRetries; retries++ {
 		resp, err = c.doRequest(method, path, body)
 		if err == nil {
-			break
+			// If we are out of API tokens, sleep first. The X-RateLimit-Reset
```

Comment:
It might be worth adding a strategy that we only sleep once? That way if we run out of API tokens, sleep, then we're already out again we don't just keep sleeping `maxRetries` more times?